farm/data_handler/data_silo.py

```diff
@@ -942,3 +942,104 @@ def _split_for_qa(documents, id_index, n_splits=5, shuffle=True, random_state=No
             current_train_set = np.hstack(np.delete(splits, idx, axis=0))
 
             yield current_train_set, current_test_set
+
+
+class DataSiloForNestedCrossVal(DataSiloForCrossVal):
+    """
+    Prepare data silos to do a nested cross validation.
+
+    The outer cross validation splits all data to test and rest.
+    The inner cross validation splits the rest data to train and dev.
```

Comment:
Lets motive the inner split in the docstring by talking about variance in multiple restarts

Reply:
I added extensive documentation. :-)